### ISSUE TITLE
Speed startup by asynchronous initialization of `ReplIO`

### ISSUE THREAD

**Issue by JaroslavTulach** [enso-org/enso#8690]
Investigation of 
- https://github.com/enso-org/enso/issues/8553#issuecomment-1878502004

revealed that another reason for slowing the startup by 16% (while measurements were off) is replacement of `DumbTerminal`:

![obrazek](https://github.com/enso-org/enso/assets/26887752/ce10972a-e257-4937-a8af-74b96ee83bb6)

by proper and more capable

```java
org.jline.terminal.impl.jna.JnaTerminalProvider.sysTerminal()
```

As the terminal isn't going to needed for next 1 or 2 seconds, let's defer its initialization into own (low priority) `Thread`.

**Comment by JaroslavTulach** [comment]
By initializing in own thread, we can eliminate `Thread.currentThread().setContextClassLoader(prevClassLoader)` - as the thread will die out after performing initialization.

**Comment by JaroslavTulach** [comment]
When at it we could rewrite `ReplIO` (and maybe also `Repl`) into Java.

**Comment by jdunkerley** [comment]
Closed as duplicate of #8688
